pkg/apis/core/validation/validation_test.go

```diff
@@ -18558,48 +18587,92 @@ func testDataSourceInSpec(name, kind, apiGroup string) *core.PersistentVolumeCla
+			expectedFail: false,
 		},
 		{
-			testName:  "test create from valid pvc source",
-			claimSpec: *testDataSourceInSpec("test_pvc", "PersistentVolumeClaim", ""),
+			testName:     "test create from valid pvc source",
+			newClaimSpec: *testDataSourceInSpec("test_pvc", "PersistentVolumeClaim", "", false),
+			expectedFail: false,
 		},
 		{
 			testName:     "test missing name in snapshot datasource should fail",
-			claimSpec:    *testDataSourceInSpec("", "VolumeSnapshot", "snapshot.storage.k8s.io"),
+			newClaimSpec: *testDataSourceInSpec("", "VolumeSnapshot", "snapshot.storage.k8s.io", false),
 			expectedFail: true,
 		},
 		{
 			testName:     "test missing kind in snapshot datasource should fail",
-			claimSpec:    *testDataSourceInSpec("test_snapshot", "", "snapshot.storage.k8s.io"),
+			newClaimSpec: *testDataSourceInSpec("test_snapshot", "", "snapshot.storage.k8s.io", false),
 			expectedFail: true,
 		},
 		{
-			testName:  "test create from valid generic custom resource source",
-			claimSpec: *testDataSourceInSpec("test_generic", "Generic", "generic.storage.k8s.io"),
+			testName:     "test create from valid generic custom resource source",
+			newClaimSpec: *testDataSourceInSpec("test_generic", "Generic", "generic.storage.k8s.io", false),
+			expectedFail: false,
 		},
 		{
 			testName:     "test invalid datasource should fail",
-			claimSpec:    *testDataSourceInSpec("test_pod", "Pod", ""),
+			newClaimSpec: *testDataSourceInSpec("test_pod", "Pod", "", false),
 			expectedFail: true,
 		},
+		{
+			testName:     "test setting resources limits in new pvc spec only should fail",
+			newClaimSpec: *testDataSourceInSpec("test_snapshot", "VolumeSnapshot", "snapshot1.storage.k8s.io", true),
+			oldClaimSpec: *testDataSourceInSpec("test_snapshot", "VolumeSnapshot", "snapshot.storage.k8s.io", false),
+			update:       true,
+			expectedFail: true,
+		},
+		{
+			testName:     "test setting no resources limits in both old pvc spec",
+			newClaimSpec: *testDataSourceInSpec("test_snapshot", "VolumeSnapshot", "snapshot1.storage.k8s.io", false),
+			oldClaimSpec: *testDataSourceInSpec("test_snapshot", "VolumeSnapshot", "snapshot.storage.k8s.io", false),
+			update:       true,
+			expectedFail: false,
+		},
+		{
+			testName:     "test setting resources limits in old pvc spec only",
+			newClaimSpec: *testDataSourceInSpec("test_snapshot", "VolumeSnapshot", "snapshot1.storage.k8s.io", false),
+			oldClaimSpec: *testDataSourceInSpec("test_snapshot", "VolumeSnapshot", "snapshot.storage.k8s.io", true),
+			update:       true,
+			expectedFail: false,
+		},
+		{
+			testName:     "test setting resources limits in both old and new pvc spec",
+			newClaimSpec: *testDataSourceInSpec("test_snapshot", "VolumeSnapshot", "snapshot1.storage.k8s.io", true),
+			oldClaimSpec: *testDataSourceInSpec("test_snapshot", "VolumeSnapshot", "snapshot.storage.k8s.io", true),
+			update:       true,
+			expectedFail: false,
+		},
```

Comment:
add tests for the cases mentioned above where the volumes are added/deleted.

Reply:
added a new function`TestValidateVolumesUpdate` for the test cases mentioned